nixos/modules/services/home-automation/home-assistant.nix

```diff
@@ -61,6 +62,26 @@ let
     extraComponents = oldArgs.extraComponents ++ extraComponents;
     extraPackages = ps: (oldArgs.extraPackages ps) ++ (cfg.extraPackages ps);
   }));
+
+  # Create a directory that holds all custom components
+  customComponentsDir = pkgs.buildEnv {
+    name = "home-assistant-custom-components";
+    paths = cfg.customComponents;
+  };
```

Comment:
(This was a problem with my previous implementation, the custom components dir needed to be its own derivation :+1: )